xyz-hub-service/src/main/java/com/here/xyz/hub/XYZHubRESTVerticle.java

```diff
@@ -128,13 +131,28 @@
   /**
    * The final response handler.
    */
-  private static void onResponseSent(RoutingContext context) {
+  private static void onResponseEnd(RoutingContext context) {
     final Marker marker = Api.Context.getMarker(context);
+    if (!context.response().headWritten()) {
```

Comment:
The new cancelling-mechanism is established to ensure the service won't do unnecessary calls to the upstream components (e.g. connectors) and to prevent wasting cpu-time and other resources. When the service starts to send the body to the client, all ustream-responses have been received and assembled for the client already.

Reply:
Makes sense